kolibri/core/assets/src/views/app-bar.vue

```diff
@@ -167,16 +158,42 @@
         ) {
           this.userMenuDropdownIsOpen = false;
         }
+        return event;
+      },
+      filterByRole(navItem) {
+        if (!navItem.role) {
+          // No role defined, so always show
+          return true;
+        }
+        if (navItem.role === UserKinds.COACH) {
+          return this.isCoach || this.isAdmin || this.isSuperuser;
+        }
+        if (navItem.role === UserKinds.ADMIN) {
+          return this.isAdmin || this.isSuperuser;
+        }
+        if (navItem.role === UserKinds.CAN_MANAGE_CONTENT) {
+          return this.canManageContent || this.isSuperuser;
+        }
+        if (navItem.role === UserKinds.SUPERUSER) {
+          return this.isSuperuser;
+        }
+        if (navItem.role === UserKinds.ANONYMOUS) {
+          return !this.isUserLoggedIn;
+        }
+        if (navItem.role === UserKinds.LEARNER) {
+          return this.isLearner || this.isCoach || this.isAdmin || this.isSuperuser;
+        }
```

Comment:
I disagree to some extent, as restriction of menu items by currently logged in user role is the main active component of our side navigation right now. I am not sure what the alternative is, or are you suggesting this is insufficiently general? I would rather keep it constrained at first, and then look to change the API if we have additional use cases.

Reply:
Specifically, I'm worried that baking the concept of 'role' into into the plugin API makes it much harder to evolve that concept later. We've already seen roles evolved a bit (superusers, device permissions, and class scoped coaches).

More generally, trying to schematize these items I think could could impede evolution of our design by locking us into the way we happen to do things now. For example, what if we decided that we wanted to add a number badge (like unread notifications, points) or other more complex interactions (pop-ups, etc)?

That's why I was thinking it would be good if plugins could drop in their own menu item component implementations, almost slot-style. These could either re-use a core `<k-side-nav-item>` implementation, or build their own from scratch. Similarly, visibility logic could depend on any core or plugin state...

Reply:
Yeah, as I noted in the general thread, this is not impossible to achieve. My main concern is that it makes the API more brittle due to the increased flexibility.

Changing the structure of the core vuex suddenly has implications that may not be immediately obvious (because they are referenced inside these additional side nav definitions) - if we were to do this, only allowing references to the core getters would be preferable, but that may well rob much of the flexibility you seem to want to allow.